python/cudf_polars/cudf_polars/containers/column.py

```diff
@@ -191,7 +194,18 @@ def obj_scalar(self) -> plc.Scalar:
         """
         if not self.is_scalar:
             raise ValueError(f"Cannot convert a column of length {self.size} to scalar")
-        return plc.copying.get_element(self.obj, 0)
+        if self._obj_scalar is None:
+            self._obj_scalar = plc.copying.get_element(self.obj, 0, stream=stream)
+            # We synchronize the stream here, to avoid a race condition if
+            # Column.obj_scalar() is ever called with different streams.
+            # We need to ensure that the result is valid on *all* streams
```

Comment:
I don't think this true even with this stream sync

Reply:
Because of what you mentioned here: https://github.com/rapidsai/cudf/pull/20291#discussion_r2456832920. If we make the scalar on stream A and we call `obj_scalar` on stream B. This stream sync is really `stream_b.synchronize()` so we are waiting on stream B to sync but the scalar was never produced on B.

Reply:
All good.

> Is it true that the column self.obj will be created before it's used by get_element?

IIUC, that's definitely required by CUDA, but I believe it's up to us to ensure that that's actually the case. And maybe "valid on" is a better phrase than "created".

https://github.com/rapidsai/cudf/blob/fd07e018c85c3053242b0e0e19939ca60d54a314/python/cudf_polars/docs/overview.md#L618 has a bit of information on what we expect in cudf-polars. Our primary rule is that any data in a `cudf_polars.DataFrame` *must* be valid on the stream passed in to that `DataFrame`. Most (but not all!) of our operations involve a `plc.Column` that's taken from a `cudf_polars.DataFrame`, and then we do stuff on `df.stream`, and so we're good to go.